packages/features/data-table/components/DataTableToolbar.tsx

```diff
@@ -30,56 +30,34 @@ const Root = forwardRef<HTMLDivElement, DataTableToolbarProps>(function DataTabl
   );
 });
 
-interface SearchBarProps<TData> {
-  table: Table<TData>;
-  searchKey?: string;
-  onSearch?: (value: string) => void;
+interface SearchBarProps {
   className?: string;
 }
 
-function SearchBarComponent<TData>(
-  { table, searchKey, onSearch, className }: SearchBarProps<TData>,
-  ref: React.Ref<HTMLInputElement>
-) {
-  const [searchTerm, setSearchTerm] = useState("");
-  const debouncedSearchTerm = useDebounce(searchTerm, 500);
-
-  useEffect(() => {
-    onSearch?.(debouncedSearchTerm);
-  }, [debouncedSearchTerm, onSearch]);
-
-  if (onSearch) {
-    return (
-      <Input
-        ref={ref}
-        className={`max-w-48 ${className ?? ""}`}
-        placeholder="Search"
-        value={searchTerm}
-        onChange={(event) => setSearchTerm(event.target.value.trim())}
-      />
-    );
-  }
-
-  if (!searchKey) {
-    console.error("searchKey is required if onSearch is not provided");
-    return null;
-  }
+function SearchBarComponent({ className }: SearchBarProps, ref: Ref<HTMLInputElement>) {
+  const { searchTerm, setSearchTerm } = useDataTable();
+  const { t } = useLocale();
+  const [localValue, setLocalValue] = useState(searchTerm);
+
+  const handleSearchChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+    setLocalValue(value);
+    setSearchTerm(value);
```

Comment:
we set local value to immediately update the <input> element, but use setSearchTerm to set debounced value to DataTable context.